### PR TITLE
Added and_masks and or_masks utilities

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #131012
* #131013
* __->__ #131073
* #130904
* #130871

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/nn/attention/flex_attention.py [hunk 1]
```diff
@@ -476,6 +485,33 @@ def _convert_mask_to_block_mask(
...
+    def or_mask(b, h, q_idx, kv_idx):
+        result = b.new_zeros((), dtype=torch.bool)
+        for mask in mask_mods:
+            result = result | mask(b, h, q_idx, kv_idx)
+        return result
+
+    return or_mask
+
+
+def and_masks(*mask_mods: _mask_mod_signature) -> _mask_mod_signature:
```
Comment (drisspg) [inline torch/nn/attention/flex_attention.py:502]: should we check if the mod type is right?

**Review by drisspg — APPROVED** [review]

File: torch/nn/attention/flex_attention.py (hunk 1 above)
Comment (Chillee) [reply torch/nn/attention/flex_attention.py:502]: Meh, I actually don't really want to use that function much 🤔 I kinda want to remove it. I don't feel like checking the signature for how many positional args it has as a safety check is very pythonic 🤔

**Pull request closed without merging**